types/es6-shim/index.d.ts

```diff
@@ -596,7 +595,7 @@ declare var Set: SetConstructor;
 
 interface WeakMap<K extends object, V> {
     delete(key: K): boolean;
-    get(key: K): V;
+    get(key: K): V | undefined;
     has(key: K): boolean;
     set(key: K, value?: V): WeakMap<K, V>;
```

Comment:
Either `set` should not have an optional `value`, or the callback to `forEach` should type value as `V | undefined`. I would err on the side of the lib definition in TypeScript and remove the optionality of `value` in `set`.